x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table.tsx

```diff
@@ -505,6 +511,40 @@ const AlertsTable: React.FunctionComponent<AlertsTableProps> = (props: AlertsTab
             rowHeightsOptions={props.rowHeightsOptions}
             onColumnResize={onColumnResize}
             ref={dataGridRef}
+            renderCustomGridBody={
+              props.dynamicRowHeight
+                ? ({ visibleColumns: _visibleColumns, Cell }) => {
```

Comment:
If possible, I'd strongly suggest **at minimum** memoizing callback, or if possible, defining it as a static component.

Example of a memoized callback within a component (better):

```tsx
const Example = () => {
  const RenderCustomGridBody: EuiDataGridProps['renderCustomGridBody'] =
    useCallback(({ visibleColumns: _visibleColumns, Cell }) => {
      // ... etc
    }, []); // whatever dependencies that should cause the body to **fully** unmount and remount

  return (
    <EuiDataGrid
      // ... other props
      renderCustomGridBody={props.dynamicRowHeight ? RenderCustomGridBody : undefined}
    />
  );
};
```

Example of a totally static component (best, but not always possible due to other dynamic data):

```tsx
const RenderCustomGridBody: EuiDataGridProps['renderCustomGridBody'] = ({ visibleColumns: _visibleColumns, Cell }) => {
  // ... etc
};

const Example = () => {
  return (
    <EuiDataGrid
      // ... other props
      renderCustomGridBody={props.dynamicRowHeight ? RenderCustomGridBody : undefined}
    />
  );
};

Reply:
Absolutely, the intention was to use a dedicated component 🙂